__tests__/integration/examples.js

```diff
@@ -0,0 +1,33 @@
+const fs = require('fs')
+const path = require('path')
+
+const examplesPath = path.join(__dirname, '../../examples')
+
+describe('examples', () => {
+  const SUT = require('../../src/index.js')
+
+  // Declare tests automatically depending on the existing folders within ./examples
+  // Use sync method in order to declare test cases synchronously
+  const examples = fs.readdirSync(examplesPath)
+
+  examples.forEach(exampleName => {
+    it(
+      `${exampleName} example should return the list of perfect width for its image and contexts`,
+      async () => {
+        expect.assertions(1)
+
+        const configurationFile = path.join(
+          examplesPath,
+          `${exampleName}/configuration.js`,
+        )
+        const resultsFile = path.join(examplesPath, `${exampleName}/result.js`)
+
+        const config = require(configurationFile)
+        const actualResults = await SUT(config)
+
+        expect(actualResults).toMatchSnapshot()
```

Comment:
These tests check if the result have changed since the last time it ran. That's what the snapshots stand for. To update it, one need to run `npm run test:examples -- -u`.

For now, I've checked the perfect image widths because the final srcset is not yet calculated. But that would be the end goal here.